tests/datavolume_test.go

```diff
@@ -811,6 +811,108 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 		})
 	})
 
+	Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:component][test] CDI Import from HTTP/S3", func() {
+
+		var (
+			dataVolume              *cdiv1.DataVolume
+			err                     error
+			originalImageName       = "cirros-qcow2.img"
+			testImageName           = "cirros-qcow2-1990.img"
+			tinyCoreIsoRateLimitURL = fmt.Sprintf("http://cdi-file-host.%s:82/cirros-qcow2-1990.img", f.CdiInstallNs)
+		)
+
+		BeforeEach(func() {
+			By("Prepare the file")
+			fileHostPod, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, utils.FileHostName, "name="+utils.FileHostName)
+			_, _, err = f.ExecCommandInContainerWithFullOutput(fileHostPod.Namespace, fileHostPod.Name, "http",
+				"/bin/sh",
+				"-c",
+				"cp /tmp/shared/images/"+originalImageName+" /tmp/shared/images/"+testImageName)
+			Expect(err).To(BeNil())
+		})
+
+		AfterEach(func() {
+			By("Delete DV")
+			err = utils.DeleteDataVolume(f.CdiClient, f.Namespace.Name, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Cleanup the file")
+			fileHostPod, err := utils.FindPodByPrefix(f.K8sClient, f.CdiInstallNs, utils.FileHostName, "name="+utils.FileHostName)
+			_, _, err = f.ExecCommandInContainerWithFullOutput(fileHostPod.Namespace, fileHostPod.Name, "http",
+				"/bin/sh",
+				"-c",
+				"rm -f /tmp/shared/images/"+testImageName)
+			Expect(err).To(BeNil())
+
+			Eventually(func() bool {
+				_, err := f.K8sClient.CoreV1().PersistentVolumeClaims(f.Namespace.Name).Get(dataVolume.Name, metav1.GetOptions{})
+				if k8serrors.IsNotFound(err) {
+					return true
+				}
+				return false
+			}, timeout, pollingInterval).Should(BeTrue())
+		})
+
+		It("[test_id:1990] CDI Data Volume - file is removed from http server while import is in progress", func() {
+			dvName := "import-file-removed"
+			By(fmt.Sprintf("Creating new datavolume %s", dvName))
+			dv := utils.NewDataVolumeWithHTTPImport(dvName, "500Mi", tinyCoreIsoRateLimitURL)
+			dataVolume, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dv)
+			Expect(err).ToNot(HaveOccurred())
+
+			phase := cdiv1.ImportInProgress
+			By(fmt.Sprintf("Waiting for datavolume to match phase %s", string(phase)))
+			err = utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, phase, dataVolume.Name)
+			if err != nil {
```

Comment:
WaitForDataVolumePhase returns a message like: "DataVolume %s not in phase %s within %v", dataVolumeName, phase, timeout as part of the err. So you can just do Expect(err).ToNot(HaveOccurred()) and it will print what you are doing here.

Reply:
nice hint, removed